Source/WTF/wtf/OSObjectPtr.h

```diff
@@ -26,42 +26,51 @@
 #pragma once
 
 #include <os/object.h>
+#include <wtf/Forward.h>
+#include <wtf/HashFunctions.h>
+#include <wtf/HashTraits.h>
 #include <wtf/StdLibExtras.h>
 
 // Because ARC enablement is a compile-time choice, and we compile this header
 // both ways, we need a separate copy of our code when ARC is enabled.
 #if __has_feature(objc_arc)
 #define adoptOSObject adoptOSObjectArc
-#define retainOSObject retainOSObjectArc
-#define releaseOSObject releaseOSObjectArc
-#define OSObjectPtr OSObjectPtrArc
 #endif
 
 namespace WTF {
 
-template<typename> class OSObjectPtr;
-template<typename T> OSObjectPtr<T> adoptOSObject(T) WARN_UNUSED_RETURN;
-
-template<typename T> static inline void retainOSObject(T ptr)
-{
 #if __has_feature(objc_arc)
-    UNUSED_PARAM(ptr);
+struct ARCEnabled : std::true_type { };
```

Comment:
Fine to land as-is, but this probably belongs somewhere more fundamental, like TypeTraits.h.